lms/djangoapps/certificates/views/webview.py

```diff
@@ -252,6 +253,17 @@ def _update_course_context(request, context, course, course_key, platform_name):
                                                               '{partner_short_name}.').format(
             partner_short_name=context['organization_short_name'],
             platform_name=platform_name)
+        
+    # Translators: this text describes the course duration
+    course_duration = CourseOverview.get_from_id(course.id).effort
```

Comment:
Do we need processing here possible exceptions?
What can happen if course overview record was deleted?

Reply:
I think no.
If `course_duration is None`, then the line will not be in the certificate.
It doesn't throw an Exception.

Reply:
I ask you about what will be if `CourseOverview.get_from_id(course.id)` returns `None`?

Reply:
I can't imagine a situation where `CourseOverview.get_from_id (course.id) == None`
The course object passed to the method is validated on [line 516](https://github.com/raccoongang/edx-platform/pull/2220/files#diff-f83bab3069b5fdf92fc009745a05605aR516) - it means `course.id` will be correct
The `CourseOverview` string associated with the course is created with each course -- therefore `CourseOverview.get_from_id (course.id)` cannot be None.


`\CourseOverview.get_from_id (course.id).effort` can be `None` -- `effort = TextField (null = True)`
but in this case on the [template](https://github.com/raccoongang/edx-platform/pull/2220/files#diff-f83bab3069b5fdf92fc009745a05605aR259) this line will not show.

Reply:
Ok. Thank you for the explanation.